packages/serialization/json/src/jsonParseNode.ts

```diff
@@ -82,15 +86,17 @@ export class JsonParseNode implements ParseNode {
     return undefined;
   }
   public getCollectionOfObjectValues = <T extends Parsable>(
-    method: ParsableFactory<T>,
+    method: ParsableFactory<T>
   ): T[] | undefined => {
-    return this._jsonNode ? (this._jsonNode as unknown[])
-      .map((x) => new JsonParseNode(x))
-      .map((x) => x.getObjectValue<T>(method)) : undefined;
+    return this._jsonNode
```

Comment:
this spot also needs the check

Reply:
Thank you. I added a second validation. Please let me know if something else is missing.